app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt

```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+                value = uiState.entity?.clazzDesc ?: "",
+                label = stringResource(id = R.string.description).addOptionalSuffix(),
+                enabled = uiState.fieldsEnabled,
+                onValueChange = {
+                    onClazzChanged(
+                        uiState.entity?.shallowCopy {
+                            clazzDesc = it
+                        }
+                    )
+                }
+            )
+        }
+
+        item {
+            UstadTextEditField(
+                value = uiState.entity?.school?.schoolName ?: "",
+                label = stringResource(id = R.string.institution),
+                enabled = uiState.fieldsEnabled,
+                onClick = onClickSchool,
+                onValueChange = {}
+            )
+        }
+
+        item {
+            UstadDateEditTextField(
+                value = uiState.entity?.clazzStartTime ?: 0,
+                label = stringResource(id = R.string.start_date),
+                error = uiState.clazzStartDateError,
+                enabled = uiState.fieldsEnabled,
+                timeZoneId = uiState.entity?.clazzTimeZone ?: "UTC",
+                onValueChange = {
+                    onClazzChanged(
+                        uiState.entity?.shallowCopy {
+                            clazzStartTime = it
+                        }
+                    )
+                }
+            )
+        }
+
+        item {
+            UstadDateEditTextField(
+                value = uiState.entity?.clazzEndTime ?: 0,
+                label = stringResource(id = R.string.end_date).addOptionalSuffix(),
+                error = uiState.clazzEndDateError,
+                enabled = uiState.fieldsEnabled,
+                timeZoneId = uiState.entity?.clazzTimeZone ?: "UTC",
+                onValueChange = {
+                    onClazzChanged(
+                        uiState.entity?.shallowCopy {
+                            clazzEndTime = it
+                        }
+                    )
+                }
+            )
+        }
+
+        item {
+            UstadTextEditField(
+                value = uiState.entity?.clazzTimeZone ?: "",
+                label = stringResource(id = R.string.timezone),
+                enabled = uiState.fieldsEnabled,
+                onClick = { onClickTimezone() },
+                onValueChange = {}
+            )
+        }
+
+        item {
+            Spacer(modifier = Modifier.height(20.dp))
+        }
+
+        item {
+            Text(text = stringResource(id = R.string.course_blocks))
+        }
+
+        item {
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickAddCourseBlock()
+                },
+                text = { Text(stringResource(id = R.string.add_block)) },
+            )
+        }
+
+        items (
+            items = uiState.courseBlockList
+        ) { courseBlock ->
+
+            val CONTENT_ENTRY_TYPE_ICON_MAP = mapOf(
```

Comment:
This map is a constant - so let's move it outside of the composable function (for performance). We don't need to make a new copy of this map every time the function runs, and then repeat that for each block.